src/cli.ts

```diff
@@ -144,43 +144,56 @@ function runWithPaths(paths: Array<string>, options: CLIOptions, callback: ((err
   let errors: Array<Error> = [];
   let pending = paths.slice();
 
-  function processPath(path: string): void {
-    stat(path, (err, info) => {
-      if (err) { errors.push(err); }
-      else if (info.isDirectory()) {
-        processDirectory(path);
-      } else {
-        processFile(path);
+  async function processPath(path: string): Promise<void> {
+    try {
+      const info = await stat(path);
+
+      if (info.isDirectory()) {
+        await processDirectory(path);
+        return processNext();
```

Comment:
I'd make this `await processNext();` since we don't actually want to propagate any return value.